fiftyone/operators/decorators.py

```diff
@@ -93,3 +96,116 @@ def dir_state(dirpath):
         state ^= hash(os.path.getmtime(os.path.dirname(p)))
 
     return state
+
+
+def execution_cache(
+    ttl=90,
+    link_to_dataset=True,
+    key_fn=None,
+    store_name=None,
```

Comment:
Idea: add a param for including `ctx` in the cache key. When `True` the cache key would include `ctx.to_dict()`. This would create new entries for various permutations of the context though.

Reply:
Should we make ttl optional and allow only dataset link based eviction?